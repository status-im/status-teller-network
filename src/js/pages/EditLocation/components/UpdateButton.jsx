import React, {Component} from 'react';
import {Row, Col, Button} from 'reactstrap';
import PropTypes from 'prop-types';
import {withNamespaces} from 'react-i18next';

class UpdateButton extends Component {
  render() {
    const t = this.props.t;
    return (
      <Row className="mt-5">
        <Col xs={12} className="text-center">
          <Button color="primary" onClick={this.props.onClick} disabled={this.props.disabled}>{t('updateUser.button')}</Button>
        </Col>
      </Row>
    );
  }
}

UpdateButton.propTypes = {
  t: PropTypes.func,
  onClick: PropTypes.func,
  disabled: PropTypes.bool
};

export default withNamespaces()(UpdateButton);
