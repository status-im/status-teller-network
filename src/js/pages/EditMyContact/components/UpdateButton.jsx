import React, {Component} from 'react';
import {Row, Col, Button} from 'reactstrap';
import PropTypes from 'prop-types';
import {withTranslation} from 'react-i18next';

class UpdateButton extends Component {
  render() {
    const t = this.props.t;
    return (
      <Row className="mt-4">
        <Col xs={12} className="text-center">
          <Button color="primary" onClick={this.props.onClick} disabled={this.props.disabled}>{t('updateUser.button')}</Button>
        </Col>
      </Row>
    );
  }
}

UpdateButton.propTypes = {
  t: PropTypes.func,
  onClick: PropTypes.func,
  disabled: PropTypes.bool
};

export default withTranslation()(UpdateButton);
