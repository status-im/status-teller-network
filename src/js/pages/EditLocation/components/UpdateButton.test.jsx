import React from "react";
import { shallow } from "enzyme";

import UpdateButton from "./UpdateButton";

describe('UpdateButton', () => {
  it('should render correctly when disabled', () => {
    const component = shallow(<UpdateButton disabled={false} onClick={() => undefined}/>);
  
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when enabled', () => {
    const component = shallow(<UpdateButton disabled={true} onClick={() => undefined}/>);
  
    expect(component).toMatchSnapshot();
  });
});
