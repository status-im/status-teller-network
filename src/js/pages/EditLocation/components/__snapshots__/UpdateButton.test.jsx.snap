// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateButton should render correctly when disabled 1`] = `
<Row
  className="mt-5"
  tag="div"
>
  <Col
    className="text-center"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs={12}
  >
    <Button
      color="primary"
      disabled={false}
      onClick={[Function]}
      tag="button"
    />
  </Col>
</Row>
`;

exports[`UpdateButton should render correctly when enabled 1`] = `
<Row
  className="mt-5"
  tag="div"
>
  <Col
    className="text-center"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs={12}
  >
    <Button
      color="primary"
      disabled={true}
      onClick={[Function]}
      tag="button"
    />
  </Col>
</Row>
`;
